packages/backend/src/plugins/catalog.ts

```diff
@@ -15,18 +15,19 @@
  */
 
 import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
-import { EntityProvider } from '@backstage/plugin-catalog-node';
 import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
 import { Router } from 'express';
 import { PluginEnvironment } from '../types';
+import createCatalogEventBasedProviders from './catalogEventBasedProviders';
 
 export default async function createPlugin(
   env: PluginEnvironment,
-  providers?: Array<EntityProvider>,
 ): Promise<Router> {
+  const providers = await createCatalogEventBasedProviders(env);
   const builder = await CatalogBuilder.create(env);
   builder.addProcessor(new ScaffolderEntitiesProcessor());
-  builder.addEntityProvider(providers ?? []);
+  env.eventBroker.subscribe(providers);
+  builder.addEntityProvider(providers);
```

Comment:
Maybe at this point, we could even consider that the event-based providers will just be registered here without the need for the separate `catalogEventBasedProviders.ts` file. The only reason to introduce it was to allow for passing the providers to the catalog and events without them knowing of each other. Of course, it is still ok to extract such things to separate files. Maybe, we shouldn't do it at the example-backend.

What do you think?

Reply:
done thanks.